### PR TITLE
Stats: Add static page modules show/hide settings

### PR DESCRIPTION
**Pull request by dognose24**
Related to #77718 

## Proposed Changes

* Add modules settings panel with the `Popover` component.
* Gate the settings panel with the feature flag `stats/modules-settings`.

## Testing Instructions

* Spin this change with the live branch link.
* Navigate to Stats > `Traffic` page.
* Append feature flag to the URL: `?flags=stats/modules-settings`.
* Ensure the settings button and panel are in line with the design.
* Ensure the toggle buttons work without changing the display of modules yet.

截圖 2023-06-04 下午10 18 31

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/blocks/stats-navigation/index.js [hunk 1]
```diff
@@ -61,6 +85,8 @@ class StatsNavigation extends Component {
 			'stats-navigation--modernized': ! isLegacy,
 		} );
 
+		const isHighlightsSettingsEnabled = config.isEnabled( 'stats/modules-settings' );
```
Comment (kangzj) [inline client/blocks/stats-navigation/index.js]:
```suggestion
		const isHighlightsSettingsEnabled = config.isEnabled( 'stats/module-settings' );
```

File: config/client.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
 	"siftscience_key",
 	"signup_url",
 	"stats/highlights-settings",
+	"stats/modules-settings",
```
Comment (kangzj) [inline config/client.json]:
```suggestion
	"stats/module-settings",
```

File: config/development.json [hunk 3]
```diff
@@ -183,6 +183,7 @@
 		"site-indicator": true,
 		"ssr/prefetch-timebox": false,
 		"stats/highlights-settings": true,
+		"stats/modules-settings": true,
```
Comment (kangzj) [inline config/development.json]:
```suggestion
		"stats/module-settings": true,
```

File: config/horizon.json [hunk 4]
```diff
@@ -121,6 +121,7 @@
 		"site-indicator": true,
 		"ssr/prefetch-timebox": true,
 		"stats/highlights-settings": false,
+		"stats/modules-settings": false,
```
Comment (kangzj) [inline config/horizon.json]:
```suggestion
		"stats/module-settings": false,
```

File: config/production.json [hunk 5]
```diff
@@ -149,6 +149,7 @@
 		"ssr/prefetch-timebox": true,
 		"ssr/sample-log-cache-misses": true,
 		"stats/highlights-settings": false,
+		"stats/modules-settings": false,
```
Comment (kangzj) [inline config/production.json]:
```suggestion
		"stats/module-settings": false,
```

File: config/stage.json [hunk 6]
```diff
@@ -141,6 +141,7 @@
 		"site-indicator": true,
 		"ssr/prefetch-timebox": true,
 		"stats/highlights-settings": false,
+		"stats/modules-settings": false,
```
Comment (kangzj) [inline config/stage.json]:
```suggestion
		"stats/module-settings": false,
```

File: config/test.json [hunk 7]
```diff
@@ -99,6 +99,7 @@
 		"site-indicator": true,
 		"ssr/prefetch-timebox": true,
 		"stats/highlights-settings": false,
+		"stats/modules-settings": false,
```
Comment (kangzj) [inline config/test.json]:
```suggestion
		"stats/module-settings": false,
```

File: config/wpcalypso.json [hunk 8]
```diff
@@ -153,6 +153,7 @@
 		"site-indicator": true,
 		"ssr/prefetch-timebox": true,
 		"stats/highlights-settings": false,
+		"stats/modules-settings": false,
```
Comment (kangzj) [inline config/wpcalypso.json]:
```suggestion
		"stats/module-settings": false,
```

**Review by kangzj — CHANGES_REQUESTED** [review]
Unfortunately 'Insights', 'Ads' and probably more pages crashed for the PR:

image

File: client/blocks/stats-navigation/index.js [hunk 9]
```diff
@@ -29,6 +32,26 @@ class StatsNavigation extends Component {
 		isLegacy: PropTypes.bool,
 	};
 
+	state = {
+		isPopoverVisible: false,
```
Comment (kangzj) [inline client/blocks/stats-navigation/index.js]: IMHO we could use a more meaningful name for Popover - for example `isPageSettingsPopoverVisible` or something?

**Review by kangzj — APPROVED** [review]
:shipit:

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8027187

Thank you @dognose24 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8027187).

**Pull request merged**